arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt

```diff
@@ -174,9 +247,113 @@ final class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdate
+                } else if (!isUsingARCore) {
+                    val camera = sceneView.currentViewpointCamera.moveTo(location)
+                    sceneView.setViewpointCamera(camera)
+                }
+            }
+        }
+
+    /**
+     * This listener is added to every [LocationDataSource] used when using the [locationDataSource] property to receive
+     * heading updates.
+     *
+     * Upon receiving an updated heading, if we're not using ARCore we create a [Camera] using the updated heading value
+     * and the pitch and roll of the current Camera from [sceneView]. We then set the current Camera of the [SceneView]
+     * to be the newly created Camera.
+     *
+     * @since 100.6.0
+     */
+    private val headingChangedListener: LocationDataSource.HeadingChangedListener =
+        LocationDataSource.HeadingChangedListener {
+            if (!isUsingARCore && !it.heading.isNaN()) {
+                // Not using ARCore, so update heading on the camera directly; otherwise, let ARCore handle heading changes.
+                val currentCamera = sceneView.currentViewpointCamera
+                val camera = currentCamera.rotateTo(it.heading, currentCamera.pitch, currentCamera.roll)
+                sceneView.setViewpointCamera(camera)
+            }
+        }
+
+    /**
+     * This listener is added to every [LocationDataSource] used when using the [locationDataSource] property to receive
+     * LocationDataSource status updates.
+     *
+     * Upon receiving the status of the LocationDataSource, we check if the status indicates a failure. If so, we set the
+     * [error] property to an Exception that indicates the current error of the LocationDataSource.
+     *
+     * @since 100.6.0
+     */
+    private val locationDataSourceStatusChangedListener: LocationDataSource.StatusChangedListener =
+        LocationDataSource.StatusChangedListener {
+            if (it.status == LocationDataSource.Status.FAILURE) {
+                error = Exception(locationDataSource?.error)
+            }
+        }
+
+    /**
+     * The data source used to get device location.  Used either in conjunction with ARCore data or when ARCore is not
+     * present or not being used.
+     *
+     * @since 100.6.0
+     */
+    var locationDataSource: LocationDataSource? = null
+        set(value) {
+            if (value == null) {
+                field?.stop()
+                field?.removeLocationChangedListener(locationChangedListener)
+                field?.removeHeadingChangedListener(headingChangedListener)
+                field?.removeStatusChangedListener(locationDataSourceStatusChangedListener)
+            }
+
+            field = value
+
+            value?.addLocationChangedListener(locationChangedListener)
+            value?.addHeadingChangedListener(headingChangedListener)
+            value?.addStatusChangedListener(locationDataSourceStatusChangedListener)
+            resetTracking()
```

Comment:
If Tracking hasn't started yet, you don't want to reset tracking.  See [L252](https://github.com/Esri/arcgis-runtime-toolkit-android/blob/ef3054ab8b79a6a64e06b53ef981f54d2f77d41b/arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt#L252)